test/client_test.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+import ray
+import sys
+import pytest
+
+
+# NOTE: These client tests run a gateway locally, which I don't think
+# makes any sense in any real-world scenario. A true Client test would
+# require multiple machines with a network configured more realistically.
+@pytest.fixture
+class ClientTest():
+    def tearDown(self):
+        ray.worker.cleanup()
+
+    def testBasicClient(self):
+        ray.init(driver_mode=ray.CLIENT_MODE)
```

Comment:
Calling `ray.init(driver_mode=ray.CLIENT_MODE)` should not start a Ray cluster (raylet, object store, etc), right? Are those processes being started somewhere? I would just expect it to start a lightweight external client which connects to an existing gateway.

I wouldn't expect `ray.init(driver_mode=ray.CLIENT_MODE)` to make any sense without also passing in a gateway address. Or am I misunderstanding the semantics?

Reply:
These tests just make sure none of the most basic pathways are broken. If no redis address is specified, the various head node processes are started locally and the client connects to this local head node. There are some comments on this use case that say this is useless except in testing.

If it's too confusing, we can ban ```driver_mode=ray.CLIENT_MODE``` if a ```redis``` address isn't specified.